File: Visual Studio LIghtswitch 2012/Chapter14/HelpDeskCS/HelpDeskCS/Client/UserCode/CreateNewEngineer.cs
using System;
using System.Linq;
using System.IO;
using System.IO.IsolatedStorage;
using System.Collections.Generic;
using Microsoft.LightSwitch;
using Microsoft.LightSwitch.Framework.Client;
using Microsoft.LightSwitch.Presentation;
using Microsoft.LightSwitch.Presentation.Extensions;
using System.ComponentModel;

namespace LightSwitchApplication
{
    public partial class CreateNewEngineer
    {
        partial void CreateNewEngineer_InitializeDataWorkspace(global::System.Collections.Generic.List<global::Microsoft.LightSwitch.IDataService> saveChangesTo)
        {
            // Write your code here.
            this.EngineerProperty = new Engineer();
        }

        partial void CreateNewEngineer_Saved()
        {
            // Write your code here.
            this.Close(false);
            Application.Current.ShowDefaultScreen(this.EngineerProperty);
        }


        //Listing 7-21. Using PropertyChanged on a new Ddata Sscreen 
        partial void CreateNewEngineer_Created()
        {
            Microsoft.LightSwitch.Threading.Dispatchers.Main.BeginInvoke(() =>
            {
                ((INotifyPropertyChanged)this.EngineerProperty).PropertyChanged +=
                    EngineerFieldChanged;                                             
            });

            //Set the initial visibility here
            this.FindControl("SecurityGroup").IsVisible = EngineerProperty.SecurityVetted;
        }

        private void EngineerFieldChanged(object sender, PropertyChangedEventArgs e)
        {
            if (e.PropertyName == "SecurityVetted")                                   
            {
                this.FindControl("SecurityGroup").IsVisible =
                    EngineerProperty.SecurityVetted;
            }
        }

    }
}

File: Visual Studio LIghtswitch 2012/Chapter14/HelpDeskVB/HelpDeskVB/Client/UserCode/CreateNewEngineer.vb
Imports System.ComponentModel

Namespace LightSwitchApplication

    Public Class CreateNewEngineer

        Private Sub CreateNewEngineer_InitializeDataWorkspace(ByVal saveChangesTo As Global.System.Collections.Generic.List(Of Global.Microsoft.LightSwitch.IDataService))
            ' Write your code here.
            Me.EngineerProperty = New Engineer()
        End Sub


        Private Sub CreateNewEngineer_Saved()
            ' Write your code here.
            Me.Close(False)
            Application.Current.ShowDefaultScreen(Me.EngineerProperty)
        End Sub


        'Listing 7-21. Using PropertyChanged on a New Ddata Sscreen 
        Private Sub CreateNewEngineer_Created()
            Microsoft.LightSwitch.Threading.Dispatchers.Main.BeginInvoke(
                Sub()
            AddHandler DirectCast(
              Me.EngineerProperty, INotifyPropertyChanged
            ).PropertyChanged, AddressOf EngineerFieldChanged
        End Sub)

            'Set the initial visibility here
            Me.FindControl("SecurityGroup").IsVisible =
                EngineerProperty.SecurityVetted
        End Sub

        Private Sub EngineerFieldChanged(
            sender As Object, e As PropertyChangedEventArgs)
            If e.PropertyName = "SecurityVetted" Then
                Me.FindControl("SecurityGroup").IsVisible =
                   EngineerProperty.SecurityVetted
            End If
        End Sub


    End Class

End Namespace